lib/qb_integration/services/line.rb

```diff
@@ -36,23 +36,48 @@ def build_from_line_items(account = nil)
         line_items.each do |line_item|
           line = create_model
 
-          price = line_item["price"]
-          quantity = line_item["quantity"]
-
-          line.amount = (quantity * price)
-          line.description = line_item["name"]
+          unless item_found = item_service.find_or_create_by_sku(line_item, account)
+            sku = line_item[:product_id] if line_item[:sku].to_s.empty?
+            sku = line_item[:sku] if sku.to_s.empty?
+            raise RecordNotFound.new "QuickBooks record not found for product: #{sku}"
+          end
 
-          line.sales_item! do |sales_item|
-            unless item_found = item_service.find_or_create_by_sku(line_item, account)
-              sku = line_item[:product_id] if line_item[:sku].to_s.empty?
-              sku = line_item[:sku] if sku.to_s.empty?
-              raise RecordNotFound.new "QuickBooks record not found for product: #{sku}"
+          if item_found.type = "Group"
+            # Currently, the ruckus QuickBooks gem we use doesn't allow for adding bundles to sales receipts
+            raise UnsupportedException.new "Bundled Item Present: FlowLink does not support adding bundled items to Sales Receipts at this time. Please contatct a FlowLink representative for more information."
+            
+            # line.group_line_detail! do |detail|
+            #   detail.id = item_found.id
+            #   detail.group_item_ref = Quickbooks::Model::BaseReference.new(item_found.name, value: item_found.id)
+            #   detail.quantity = line_item["quantity"]
+          
+            #   item_found.item_group_details.line_items.each do |group_line|
+            #     g_line_item = create_model
+            #     group_item_found = item_service.find_by_id(group_line.id)
+
+            #     g_line_item.amount = group_line.quantity.to_i * group_item_found.unit_price.to_f
+
+            #     g_line_item.sales_item! do |gl|
+            #       gl.item_id = group_line.id
+            #       gl.quantity = group_line.quantity.to_i
+            #       gl.unit_price = group_item_found.unit_price.to_f
+            #     end
+          
+            #     detail.line_items << g_line_item
+            #   end
+            # end
```

Comment:
(collapses to the floor at the sight of so many lines of commented code)

Reply:
😂